bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1132,26 +1363,70 @@ private static boolean checkObjectEquivalency(BType sourceType, BObjectType targ
             return false;
         }
 
+
+        String targetTypeModule = Optional.ofNullable(targetType.getPackage()).map(BPackage::getName).orElse("");
+        String sourceTypeModule = Optional.ofNullable(sourceObjectType.getPackage()).map(BPackage::getName).orElse("");
```

Comment:
We should consider the org name too right? Not sure whether the version should be considered though

Reply:
We should consider all 3 right? Two different versions of the same module can be considered as two different modules IMO, when considering visibility regions.

Will fix, thanks.

Reply:
Came across https://github.com/ballerina-platform/ballerina-lang/issues/25199 when trying to fix this. Both fixed in 8d23769.